client/src/panels/panel_declarations/results/result_flat_table.js

```diff
@@ -74,56 +74,68 @@ const subject_link = (node) => (
   </span>
 );
 
-
 const indicator_table_from_list = (indicator_list) => {
-  const column_names = {
-    cr_or_program: text_maker("cr_or_program"),
-    indicator: text_maker("indicator"),
-    target: text_maker("target"),
-    target_result: text_maker("target_result"),
-    date_to_achieve: text_maker("date_to_achieve"),
-    status: text_maker("status"),
+  const subject_link_map = _.chain(indicator_list)
+    .map(ind => [ind.parent_node.data.name, subject_link(ind.parent_node)])
+    .fromPairs()
+    .value();
+  const indicator_id_map = _.chain(indicator_list)
+    .map(ind => [ind.indicator.name, `#indicator/${ind.indicator.id}`])
+    .fromPairs()
+    .value();
+  const column_configs = {
+    cr_or_program: {
+      index: 0,
+      header: text_maker("cr_or_program"),
+      is_sortable: true,
+      is_searchable: true,
+      formatter: (value) => subject_link_map[value],
+    },
+    indicator: {
+      index: 1,
+      header: text_maker("indicator"),
+      is_sortable: true,
+      is_searchable: true,
+      formatter: (value) => <a href={indicator_id_map[value]}> {value} </a>,
+    },
+    target: {
+      index: 2,
+      header: text_maker("target"),
+    },
+    target_result: {
+      index: 3,
+      header: text_maker("target_result"),
+    },
+    date_to_achieve: {
+      index: 4,
+      header: text_maker("date_to_achieve"),
+      is_sortable: true,
+      formatter: (val) => _.isDate(val) ?
+        `${months[val.getMonth() + 1].text} ${val.getFullYear()}` : val,
+    },
+    status: {
+      index: 5,
+      header: text_maker("status"),
+      is_sortable: true,
+      formatter: (value) => <Fragment>
+        <span aria-hidden="true" className="copyable-hidden">{result_statuses[value].text}</span>
+        {window.innerWidth < breakpoints.mediumDevice ? status_icons[value] : large_status_icons[value]}
```

Comment:
Should use React `MediaQuery` components so this automatically re-renders to the appropriate size when the screen resizes. Also `breakpoints.mediumDevice` is undefined, it's either `minMediumDevice` or `maxMediumDevice` (depending if you're doing a `<MediaQuery minWidth={minMediumDevice}>` or `<MediaQuery maxWidth={maxMediumDevice}>` respectively).

Looking at it now, that stuff's super poorly documented in the InfoBase, my bad. If you want to know why breakpoints are stupid like this ask some other time (long story short: they're a nightmare).

Reply:
I played around with it and tested it around, I think this one looks better with `large_status_icons` at all times. Thanks for letting me know with breakpoints though and putting this one on spotlight!